_field-types/flat-object.md

```diff
@@ -0,0 +1,218 @@
+{: .note}
+
+Flat objects provide the following advantages:
+
+- Efficient reads: Fetching performance is similar to that of a keyword field.
+- Memory efficiency: Storing the entire complex JSON object in one field without indexing all its subfields reduces the number of fields in an index. 
+- Space efficiency: OpenSearch does not create an inverted index for subfields in flat objects, thereby saving space. 
+- Compatibility for migration: You can migrate your data from systems that support similar flat types to OpenSearch.
+
+Mapping a field as flat object applies when a field and its subfields are mostly read and not used as a search criteria because the subfields are not indexed. Flat objects are useful for objects with a large number of fields or when you don't know the keys in advance.
+
+Flat objects support exact match queries with and without dot path. For a complete list of supported query types, see [Supported queries](#supported-queries).
+
+Searching for a specific value of a nested field in a document may be inefficient because it may require a full scan of the index, which is an expensive operation.
+{: .note}
+
+Flat objects do not support:
+
+- Type-specific parsing.
+- Numerical operations, such as numerical comparison and numerical sorting.
+- Text analysis.
+- Highlighting.
+- Aggregations of subfields using dot notation.
+- Filtering by subfields.
+
+## Supported queries
+
+The flat object field type supports the following queries:
+
+- [Term]({{site.url}}{{site.baseurl}}/query-dsl/term#term) 
+- [Terms]({{site.url}}{{site.baseurl}}/query-dsl/term#terms) 
+- [Terms set]({{site.url}}{{site.baseurl}}/query-dsl/term#terms-set)  
+- [Prefix]({{site.url}}{{site.baseurl}}/query-dsl/term#prefix) 
+- [Range]({{site.url}}{{site.baseurl}}/query-dsl/term#range) 
+- [Match]({{site.url}}{{site.baseurl}}/query-dsl/full-text/#match) 
+- [Multi-match]({{site.url}}{{site.baseurl}}/query-dsl/full-text/#multi-match) 
+- [Query string]({{site.url}}{{site.baseurl}}/query-dsl/full-text/#query-string) 
+- [Simple query string]({{site.url}}{{site.baseurl}}/query-dsl/full-text/#simple-query-string) 
+- [Exists]({{site.url}}{{site.baseurl}}/query-dsl/term#exists) 
+
+## Limitations
+
+The following limitations apply to flat objects in the 2.7 release:
+
+- Flat objects do not support open parameters.
+- Painless scripting and wildcard queries are not supported for retrieving values of subfields.
+
+This functionality is planned for a future release.
+
+## Using flat objects 
+
+The following example illustrates mapping a field as a flat object, indexing documents with flat object fields, and searching for leaf values of the flat object in those documents.
+
+Only root fields of a document can be defined as a flat objects. You cannot define an object that is part of another JSON object as a flat object because when a flat object it is flattened to a string, the nested architecture of the leaves is lost.
```

Comment:
`Only the root field of a document can be defined as a flat object. ` Better to keep the singular form.